### PR TITLE
Added data migration to de-duplicate data from category_boards table(…

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
This is essentially a cherry-pick of https://github.com/mattermost/focalboard/pull/4614

This was committed to focalboard after cutoff for mono-repo.

#### Ticket Link

https://github.com/mattermost/focalboard/issues/4593

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by wiggin77** [comment]
@sbishel there is a reference to Sqlite in there somewhere, causing CI error.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: server/boards/services/store/sqlstore/data_migrations.go [hunk 1]
```diff
@@ -793,3 +794,94 @@ func (s *SQLStore) getCollationAndCharset(tableName string) (string, string, err
...
+	if err := row.Scan(&count); err != nil {
+		s.logger.Error("Error occurred reading number of duplicate records in category_boards table", mlog.Err(err))
+		return false, err
+	}
+
+	return count > 0, nil
+}
+
+func (s *SQLStore) runMySQLDeDuplicateCategoryBoardsMigration() error {
+	query := "WITH duplicates AS (SELECT id, ROW_NUMBER() OVER(PARTITION BY user_id, board_id) AS rownum " +
```
[the remark this answers is not in the document]
Comment (sbishel) [reply server/boards/services/store/sqlstore/data_migrations.go:866]: @harshilsharma63 - Could you have a look...cp of https://github.com/mattermost/focalboard/pull/4614

[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/boards/services/store/sqlstore/data_migrations.go:866]: Fixed in #22606

**Pull request merged**